odl/discr/grid.py

```diff
@@ -515,6 +515,40 @@ def insert(self, index, other):
                     self.coord_vectors[idx:])
         return TensorGrid(*new_vecs)
 
+    def remove(self, axes):
```

Comment:
I don't want to make grids mutable, so `__delitem__` is not an option for me. The list analogy is a valid point, so `remove_axes` would be an alternative.